File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;
public static class SecretHandshake
{

    public static string[] Commands(int commandValue)
    {
        List<string> commandList = new List<string>();
        Dictionary<int, string> commands = new Dictionary<int, string>()
                                        {
                                            {1, "wink"},
                                            {2, "double blink"},
                                            {4, "close your eyes"},
                                            {8, "jump"}
                                        }; 
        for(int i = 1; i < 9; i *= 2 )
        {
            if ((commandValue & i) == i)
            {
                commandList.Add(commands[i]);
            }
        }
        if((commandValue & 16) == 16)
        {
            commandList.Reverse();
        }
        return commandList.ToArray();
    }
}

File: typescript/secret-handshake/secret-handshake.ts
export default class HandShake{
    private readonly commandMask: number;
    constructor(input: number){
        if(!Number.isInteger(input)){
            throw new Error('Handshake must be a number');
        } 
        this.commandMask = input;
    }
    commands(): string[]{
        let result = [];
        for(let i = 0; i < answers.length; i++){
            if((this.commandMask & Math.pow(2, i))){
              result.push(answers[i]);
            }
        }
        return (this.commandMask & 16) ? result.reverse() : result;
    }
}

const answers: string[] = ['wink', 'double blink', 'close your eyes', 'jump'];